python/pyspark/pandas/sql_formatter.py

```diff
@@ -265,7 +266,13 @@ def _convert_value(self, val: Any, name: str) -> Optional[str]:
             val._to_spark().createOrReplaceTempView(df_name)
             return df_name
         elif isinstance(val, str):
-            return lit(val)._jc.expr().sql()  # for escaped characters.
+            if is_remote():
+                # This is matched to behavior from JVM implementation.
+                # See `sql` definition from `sql/catalyst/src/main/scala/org/apache/spark/
+                # sql/catalyst/expressions/literals.scala`
+                return "'" + val.replace("\\", "\\\\").replace("'", "\\'") + "'"
+            else:
+                return lit(val)._jc.expr().sql()  # for escaped characters.
```

Comment:
```suggestion
            # This is matched to behavior from JVM implementation.
            # See `sql` definition from `sql/catalyst/src/main/scala/org/apache/spark/
            # sql/catalyst/expressions/literals.scala`
            return "'" + val.replace("\\", "\\\\").replace("'", "\\'") + "'"
```

Reply:
Actually let's remove the jvm access here then.

Reply:
Sounds good